File: snippets/java/src/main/java/develop/DurableTimers.java
package develop;

import dev.restate.sdk.Restate;
import dev.restate.sdk.common.TimeoutException;
import java.time.Duration;

public class DurableTimers {

  public void timers() {
    // <start_sleep>
    Restate.sleep(Duration.ofSeconds(10));
    // <end_sleep>
  }

  public void timeouts() {
    // <start_timer>
    try {
      Restate.serviceHandle(MyService.class)
          .call(MyService::myHandler, "Hi!")
          .await(Duration.ofSeconds(5));
    } catch (TimeoutException e) {
      // Handle the timeout error
    }
    // <end_timer>
  }
}

File: snippets/kotlin/src/main/kotlin/develop/DurableTimers.kt
package develop

import dev.restate.sdk.common.TimeoutException
import dev.restate.sdk.kotlin.*
import kotlin.time.Duration.Companion.seconds

class DurableTimers {
  suspend fun timers() {
    // <start_sleep>
    sleep(10.seconds)
    // <end_sleep>
  }

  suspend fun timeouts() {
    // <start_timer>
    try {
      toService<MyService>().request { myHandler("Hi!") }.call().await(5.seconds)
    } catch (e: TimeoutException) {
      // Handle the timeout
    }
    // <end_timer>
  }
}
